lona/client_pre_compiler.py

```diff
@@ -48,33 +69,65 @@ def get_enums(self) -> Dict[str, Dict[str, Any]]:
 
         return enums
 
+    def get_source_files(self) -> list[str]:
+        if self.server.settings.CLIENT_DEBUG:
+            return [
+                '_lona/lona.js',
+                *SOURCE_FILES,
+            ]
+
+        return [
+            '_lona/lona.js',
+        ]
+
     def compile(self) -> None:
         logger.debug('pre compiling client')
 
-        try:
-            template = self.jinja2_env.get_template('lona.js')
+        if self.server.settings.CLIENT_DEBUG:
+            logger.debug('running in debug mode')
+
+        for source_file in self.get_source_files():
+            logger.debug('pre compiling %s', source_file)
+
+            try:
+                template = self.jinja2_env.get_template(source_file)
+
+                template_context = {
+                    'DEBUG': self.server.settings.CLIENT_DEBUG,
+                    'source_files': SOURCE_FILES,
+                    'protocol': dumps(self.get_enums()),
+                    'settings': dumps(self.get_settings()),
+                }
+
+                file_content = template.render(
+                    **template_context,
+                )
+
+                path = os.path.join(
+                    self.tmp_dir.name,
+                    source_file,
+                )
 
-            template_context = {
-                'protocol': dumps(self.get_enums()),
-                'settings': dumps(self.get_settings()),
-            }
+                with open(path, 'w+') as f:
+                    f.write(file_content)
+                    f.close()
```

Comment:
This line of code may be redundant as the file will be closed by the context manager in `__exit__`

Reply:
Ack! :) fixed!

Reply:
Happy to have helped you :)
Just a fresh look

Reply:
Review is always welcome!